src/context.ts

```diff
@@ -223,10 +240,13 @@ export class NuxtIconModuleContext {
       }
     }))
 
-    if (customCollections.length) {
-      customCollections.flatMap(collection => Object.entries(collection.icons)
-        .map(([name, data]) => {
-          addIcon(collection.prefix, name, data)
+    if (includeCustomCollections && customCollections.length) {
+      customCollections.flatMap(collection => Object.keys(collection.icons)
+        .map((name) => {
+          const data = getIconData(collection, name)
+          if (data) {
+            addIcon(collection.prefix, name, data)
+          }
```

Comment:
This is the change which properly merges icon settings with collection settings. Previously, each SVG icon had all the metadata (mostly width and height) embedded in its declaration. With `IconifyJSON`, it is common to keep common values on the collection level.

```js
// SVG collection as loaded from `dir`
{
  prefix: 'svg',
  icons: {
    'nuxt-v2': { body: '...', height: 512, width: 512 },
    'nuxt-v3': { body: '...', height: 512, width: 512 },
  }
}

// _Optimized_ IconifyJSON for premium icon set
{
  prefix: 'optimized',
  icons: {
    'nuxt-v2': { body: '...' },
    'nuxt-v3': { body: '...' },
  },
  height: 512,
  width: 512,
}
```